control-plane/bfruntime.cpp

```diff
@@ -601,9 +601,15 @@ BFRuntimeGenerator::makeActionSpecs(const p4configv1::Table& table,
         for (const auto& param : action->params()) {
             auto* annotations = transformAnnotations(
                 param.annotations().begin(), param.annotations().end());
-            addActionDataField(
-                dataJson, param.id(), param.name(), true /* mandatory */,
-                false /* read_only */, makeTypeBytes(param.bitwidth()), annotations);
+            if (param.type_name().name() == "PSA_MeterColor_t") {
```

Comment:
it may be safer to check the type of the parameter as obtained from the typeMap (the canonical type).
This may not work if the type is a typedef (unless the midend actually eliminates typedefs).
You should probably add a test for this case.

Reply:
bfrt generation happens before midend.